code/game/machinery/vending/vendor_types/crew/corporate_security.dm

```diff
@@ -0,0 +1,198 @@
+
+	list("RAIL ATTACHMENTS (CHOOSE 2)", 0, null, null, null),
+	list("Red-Dot Sight", 0, /obj/item/attachable/reddot, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Reflex Sight", 0, /obj/item/attachable/reflex, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("S4 2x Telescopic Mini-Scope", 0, /obj/item/attachable/scope/mini, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Magnetic Harness", 0, /obj/item/attachable/magnetic_harness, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Laser Sight", 0, /obj/item/attachable/lasersight, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Angled Grip", 0, /obj/item/attachable/angledgrip, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Vertical Grip", 0, /obj/item/attachable/verticalgrip, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Extended Barrel", 0, /obj/item/attachable/extended_barrel, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+	list("Suppressor", 0, /obj/item/attachable/suppressor, CIVILIAN_CAN_BUY_GLASSES, VENDOR_ITEM_REGULAR),
+
+	list("SPARE EQUIPMENT", 0, null, null, null),
+	list("Handheld Flash", 2, /obj/item/device/flash, null, VENDOR_ITEM_REGULAR),
+	list("Pepper Spray", 4, /obj/item/reagent_container/spray/pepper, null, VENDOR_ITEM_REGULAR),
+	list("Stun Baton", 4, /obj/item/weapon/baton, null, VENDOR_ITEM_REGULAR),
+	list("Box of Zipties", 4, /obj/item/storage/box/zipcuffs/small, null, VENDOR_ITEM_REGULAR),
+))
+
+/obj/structure/machinery/cm_vending/clothing/corporate_security
+	name = "\improper Corporate Security Wardrobe"
+	desc = "A wardrobe containing all the clothes a Personal Protection Officer would ever need."
+	icon_state = "wardrobe_vendor"
+	vendor_theme = VENDOR_THEME_COMPANY
+	req_access = list(ACCESS_WY_SECURITY)
+	vendor_role = list(JOB_CORPORATE_BODYGUARD)
+	desc = "An automated rack hooked up to a colossal storage of Corporate Security standard-issue equipment."
+	show_points = FALSE
+
+/obj/structure/machinery/cm_vending/clothing/corporate_security/get_listed_products(mob/living/carbon/human/user)
+	return GLOB.cm_vending_clothing_corporate_security
+
+/obj/structure/machinery/cm_vending/gear/corporate_security
+	name = "\improper Corporate Security Equipment Rack"
+	desc = "A wardrobe containing all the clothes a Personal Protection Officer would ever need."
+	icon_state = "clothing"
+	vendor_theme = VENDOR_THEME_COMPANY
+	req_access = list(ACCESS_WY_SECURITY)
+	vendor_role = list(JOB_CORPORATE_BODYGUARD)
+	desc = "An automated rack hooked up to a colossal storage of Corporate Security standard-issue equipment."
+
+/obj/structure/machinery/cm_vending/gear/corporate_security/get_listed_products(mob/living/carbon/human/user)
+	if(user?.wear_id)
+		var/obj/item/card/id/id_card = user.wear_id
+		if(id_card.paygrade == PAY_SHORT_WY_SEC_TPPO)
+			return GLOB.cm_vending_gear_corporate_security_non_lethal
+	return GLOB.cm_vending_gear_corporate_security_full
```

Comment:
as mentioned in a prior discussion, I'm against the vendors morphing based on playtime no matter how low the requirement.